netket/machine/torch.py

```diff
@@ -67,7 +67,22 @@ def log_val(self, x, out=None):
         return out
 
     def der_log(self, x, out=None):
-        return NotImplementedError
+        x = _torch.tensor(x, dtype = _torch.float64)
+        parameters = list(self._module.parameters())
+        out = x.new_empty(
+                             [x.size(0), 2, sum(map(_torch.numel, parameters))],
```

Comment:
I think there is no need to recompute the number of paremeters every time with 
```python
sum(map(_torch.numel, parameters))
```
you can just call self.n_par

Reply:
similarly do we need to do 
```python
parameters = list(self._module.parameters())
```
?

Reply:
seemingly cleaned this up